app/src/main/java/com/example/travellens/fragments/ComposeFragment.java

```diff
@@ -156,6 +156,33 @@ public void onClick(View v) {
                 startActivityForResult(Intent.createChooser(intent, "Pick an image"), REQUEST_CODE_GALLERY);
             }
         });
+
+        // allows this fragments menu to behave differently than in main
+        setHasOptionsMenu(true);
+
+        // initializes the client
+        if (!Places.isInitialized()) {
+            // initialize the api with key
+            Places.initialize(getContext(), getString(R.string.google_maps_api_key));
```

Comment:
Is this something that needs to be executed just once per app run? If so you can create a class that extends Application and do it there.

Reply:
Good idea! I tested it and it seems like once per app run is the right way to do it.